types/src/proof/definition.rs

```diff
@@ -241,6 +241,66 @@ impl IntoProto for SparseMerkleProof {
     }
 }
 
+/// A proof that can be used to show that two Merkle accumulators are consistent -- the big one can
+/// be obtained by appending certain leaves to the small one. For example, at some point in time a
+/// client knows that the root hash of the ledger at version 10 is `old_root` (it could be a
+/// waypoint). If a server wants to prove that the new ledger at version `N` is derived from the
+/// old ledger the client knows, it can show the frozen subtrees that form the small accumulator
+/// and the siblings that represent the new leaves. If the client can verify that the provided
+/// frozen subtrees match the old root hash and the new root hash can be obtained by combining the
+/// frozen subtrees with the siblings, it can be convinced that the two accumulators are
+/// consistent.
+///
+/// ```text
+///                                                  new_root (N = 17~32)
+///                                                 /        \
+///                                                /          \
+///                                   old_root (10)            siblings[2]
+///                                  /             \
+///                                /                 \
+///                              /                     \
+///                            /                         \
+///                          /                             \
+///                        /                                 \
+/// frozen_subtree_roots[0]                                   o
+///                    /   \                                 / \
+///                   /     \                               /   \
+///                  o       o                             o     siblings[1]
+///                 / \     / \                           / \
+///                o   o   o   o   frozen_subtree_roots[1]   siblings[0]
+///               / \ / \ / \ / \                      / \
+///               o o o o o o o o                      o o
+/// ```
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct AccumulatorConsistencyProof {
+    /// The root hashes of the frozen subtrees that form the small accumulator.
+    frozen_subtree_roots: Vec<HashValue>,
```

Comment:
Shall we assume the client desiring for such a proof already have the subtree roots?  Because a earlier verification can yield that as a (side) result.

Reply:
That may not be possible given the current setup? For instance, if `N = 17` (there is one leaf under `siblings[2]` in the above example, the new frozen subtree will need to include this leaf, however the leaf is not known to the client.

Reply:
I think @msmouse is asking if the client already has the subtree roots, not the siblings.  If the client doesn't have the subtree roots, we can just provide any random ones we want here - those nodes don't actually prove anything if they come from the server - so they probably don't need to be in the proof.

Reply:
I don't think there is currently any API for the client to obtain the subtree roots, so I think we can put them here.

I think the client would have the root hash of the previous accumulator, and it can combine the subtree roots and see if they match the root hash, so providing random ones won't work. The verification consists of two steps:

1. Check whether the subtree roots match the old root hash.
2. Combine the subtree roots and the siblings to check if we can obtain the new root hash.

Reply:
Yeah, a thin client may not have the accu structure except for the root hash only. So a consistent proof should be two parts. The 1st part can generate known root and 1st + 2nd can generate the new root.

Reply:
Re "there's no API to get the roots": when it boots up from client_known_version==0, the siblings in the proof should help it constructing the current roots?